homeassistant/components/notify/nfandroidtv.py

```diff
@@ -133,6 +139,12 @@ def send_message(self, message="", **kwargs):
 
         data = kwargs.get(ATTR_DATA)
         if data:
+            if ATTR_ICON in data:
+                icon_file = data.get(ATTR_ICON)
+                payload[ATTR_FILENAME] = ('icon.png',
```

Comment:
Since this attribute always should be set, you can remove this here and instead just do it below, always. Just check if we should set the `icon_file` from service data here. You need to make sure it has a default value in this method too, ie:
```py
icon_file = None
data = kwargs.get(ATTR_DATA)
if data:
    icon_file = data.get(ATTR_ICON)
```